Controller/Payment/Webhook.php

```diff
@@ -0,0 +1,446 @@
+            $quoteId        = $rzpOrder->receipt;
+            $rzpOrderAmount = $rzpOrder->amount;
+            if (isset($quoteId) === false) {
+                $this->logger->info("Razorpay Webhook: Quote ID not set for Razorpay payment_id(:$paymentId)");
+                return;
+            }
+        } catch (\Razorpay\Api\Errors\Error $e) {
+            $this->logger->critical("Razorpay Webhook: fetching RZP order data(id:$rzpOrderId) "
+                                . "PaymentId:(:$paymentId) failed with error: ". $e->getMessage());
+            return;
+        } catch (\Exception $e) {
+            $this->logger->critical("Razorpay Webhook: fetching RZP order data(id:$rzpOrderId) "
+                                . "PaymentId:(:$paymentId) failed with error: ". $e->getMessage());
+            return;
+        }
+        try {
+            # fetch the related sales order and verify the payment ID with rzp payment id
+            # To avoid duplicate order entry for same quote
+            $collection = $this->objectManagement->get('Magento\Sales\Model\Order')
+            ->getCollection()
+            ->addFieldToSelect('entity_id')
+            ->addFilter('increment_id', $quoteId)
+            ->getFirstItem();
+            $salesOrder = $collection->getData();
+            if (isset($salesOrder['entity_id']) && empty($salesOrder['entity_id']) === false) {
+                $this->logger->info("Razorpay inside order already processed with webhook quoteID:" . $quoteId
+                                    ." and OrderID:".$salesOrder['entity_id']);
+                $order = $this->order->load($salesOrder['entity_id']);
+                if ($order) {
+                    if ($order->getStatus() === 'pending') {
+                        $this->checkoutSession
+                            ->setLastQuoteId($order->getQuoteId())
+                            ->setLastSuccessQuoteId($order->getQuoteId())
+                            ->clearHelperData();
+                        $this->checkoutSession->setLastOrderId($order->getId())
+                                ->setLastRealOrderId($order->getIncrementId())
+                                ->setLastOrderStatus($order->getStatus());
+                        $amountPaid = number_format($rzpOrderAmount / 100, 2, ".", "");
+                        $order->setState(static::STATUS_PROCESSING)->setStatus(static::STATUS_PROCESSING);
+                        $order->addStatusHistoryComment(
+                            __(
+                                'Actual Amount %1 of %2, with Razorpay Offer/Fee applied.',
+                                "Authroized",
+                                $order->getBaseCurrency()->formatTxt($amountPaid)
+                            )
+                        );
+                        $order->save();
+                        //update quote
+                        $quote = $this->objectManagement->get('Magento\Quote\Model\Quote')
```

Comment:
@enishant , this is not required as we are working with orders not the quotes.

Reply:
@ramth05 Removed code written for quote update